gpt_engineer/learning.py

```diff
@@ -153,21 +153,14 @@ def ask_if_can_store() -> bool:
     return can_store == "y"
 
 
-def logs_to_string(steps: List[Step], logs: DB):
+def logs_to_string(steps: List[Step], logs: DB) -> str:
     chunks = []
     for step in steps:
         chunks.append(f"--- {step.__name__} ---\n")
-        messages = json.loads(logs[step.__name__])
-        chunks.append(format_messages(messages))
+        chunks.append(logs[step.__name__])
     return "\n".join(chunks)
 
 
-def format_messages(messages: List[dict]) -> str:
-    return "\n".join(
-        [f"{message['role']}:\n\n{message['content']}" for message in messages]
-    )
-
```

Comment:
This breakes tests. See CI.

Any reason of changing it?

Reply:
Deleting `format_messages` is not the issue. That function is not used anywhere. That's why I deleted it.

The test `test_collect_learnings` breaks because in the test we're doing sth like
```
code = "this is output\n\nit contains code"
code_with_extra_stuff = "extra stuff" + json.dumps(code)
assert code in code_with_extra_stuff
```
but `json.dumps` escapes the `\n` and turns it into `\\n`. That's one of the tests that fail which iiuc don't relate to the changes in this PR.

I will fix the tests by changing `assert code in code_with_extra_stuff` to `assert json.dumps(code) in code_with_extra_stuff`